File: soundmarine_client/lib/models/track.dart
class Track {
  final String id;
  final String title;
  final int trackNumber;
  final String? albumId;
  final String? artistId;
  final String artistName;
  final int duration;
  final String? dateAdded;

  Track({
    required this.id,
    required this.title,
    required this.trackNumber,
    this.albumId,
    this.artistId,
    required this.artistName,
    required this.duration,
    this.dateAdded,
  });

  factory Track.fromJson(Map<String, dynamic> json) {
    return Track(
      id: json['id'],
      title: json['title'],
      trackNumber: json['trackNumber'],
      albumId: json['albumId'],
      artistId: json['artistId'],
      artistName: json['artistName'],
      duration: json['duration'],
      dateAdded: json['dateAdded'],
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'title': title,
    'trackNumber': trackNumber,
    'albumId': albumId,
    'artistId': artistId,
    'artistName': artistName,
    'duration': duration,
    'dateAdded': dateAdded,
  };
}

File: Soundmarine.lib/Model/Track.cs
namespace Soundmarine.lib.Model;

public class Track
{
    public string Id { get; set; }
    public string Filepath { get; set; }
    public string Title { get; set; }
    public int TrackNumber { get; set; }
    public string? AlbumId { get; set; }
    public string? ArtistId { get; set; }
    public int Duration { get; set; }

    public Track(string id,string filepath, string title, int trackNumber, string? albumId, string? artistId, int duration)
    {
        Id = id;
        Filepath = filepath;
        Title = title;
        TrackNumber = trackNumber;
        AlbumId = albumId;
        ArtistId = artistId;
        Duration = duration;
    }
}